doc/source_tech_preview/reference.rst

```diff
@@ -0,0 +1,333 @@
+
+`openstack-helm-images` provides Dockerfiles directly for all the
+non-OpenStack images.
+
+For the OpenStack images, `openstack-helm-images` contains shell scripts,
+situated in `openstack/loci/`. The `build.sh` script is a thin wrapper around
+`LOCI`. `LOCI` is the official OpenStack project to build OCI compliant
+images of OpenStack projects. It uses `docker build` to construct images from
+openstack sources and their requirements are expressed in `bindep` files
+(`bindep.txt` for rpm/apt packages, `pydep.txt` for python packages).
+The `build.sh` runs `LOCI` for the master branch. Other branches can be built
+suing `build-{branchname}.sh` where `branchname` is the name of the OpenStack
+branch (for example, `rocky`). See also :ref:`buildlociimages`.
+
+In the future, `openstack-helm-images` could theoretically add images for
+OpenStack which would be based on packages, by simply providing the appropriate
+Dockerfiles.
+
+Additionally, some images are not built in `openstack-helm-images`, and they
+are directly consumed/fetched from upstream projects official dockerfiles,
+like xrally.
+
+socok8s process
+---------------
+
+socok8s leverages the existing OSH-images code.
+
+When running the `build_images` step, the `localhost` asks the `deployer` to
+build images based on the code that was checked in on the `deployer` node
+using the `vars/manifest.yml`.
+
+For the non-LOCI images, the `suse-build-images` role invoked in the
+`build_images` step is simply running a `docker build` command.
+
+For the LOCI images, the `suse-build-images` role simply runs the command
+available in `openstack-helm-images` calling the LOCI build.
+
+OpenStack-Helm chart overrides
+==============================
+
+Helm chart values overriding principle
+--------------------------------------
+
+A helm chart installation
+(See https://helm.sh/docs/using_helm/#customizing-the-chart-before-installing )
+accepts an argument named ``--values`` or ``-f``.
+
+This argument expects the filename of a YAML file present on the
+helm client machine. It can be specified multiple times, and
+the rightmost file will take precedence.
+
+For example, in the following example, the different values of
+``socok8s-glance.yaml`` overrides would win over the existing values in
+``/tmp/glance.yaml``:
+
+.. code-block:: console
+
+   helm upgrade --install glance ./glance --namespace=openstack \
+     --values=/tmp/glance.yaml --values=/tmp/socok8s-glance.yaml
+
+OpenStack-Helm scripts
+----------------------
+
+The OpenStack-Helm project provides shell scripts to deploy the helm charts,
+with overrides per context (for example, multinode).
+
+Those shell scripts calling the helm installation include an environment
+variable, to allow users to pass extra arguments.
+
+See for example https://github.com/openstack/openstack-helm/blob/c869b4ef4a0e95272155c5d5dd893c72976753cd/tools/deployment/multinode/100-glance.sh#L49 .
+
+SUSE-ification of the OSH charts when deploying in OSH only mode
```

Comment:
Please don't use the term "SUSE-ification", it is misleading - there is no green paint added.

Reply:
"no green paint added." :-)